src/Domain/FunctionsOfTime/ReadSpecPiecewisePolynomial.cpp

```diff
@@ -199,6 +201,94 @@ void read_spec_piecewise_polynomial(
     }
   }
 }
+
+void override_functions_of_time(
+    const gsl::not_null<std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>*>
+        functions_of_time,
+    const std::string& function_of_time_file,
+    const std::map<std::string, std::string>& function_of_time_name_map) {
+  std::unordered_map<std::string,
+                     domain::FunctionsOfTime::PiecewisePolynomial<2>>
+      spec_functions_of_time_second_order{};
+  std::unordered_map<std::string,
+                     domain::FunctionsOfTime::PiecewisePolynomial<3>>
+      spec_functions_of_time_third_order{};
+  std::unordered_map<std::string,
+                     domain::FunctionsOfTime::QuaternionFunctionOfTime<3>>
+      spec_functions_of_time_quaternion{};
+
+  // Import those functions of time of each supported order
+  domain::FunctionsOfTime::read_spec_piecewise_polynomial(
+      make_not_null(&spec_functions_of_time_second_order),
+      function_of_time_file, function_of_time_name_map);
+  domain::FunctionsOfTime::read_spec_piecewise_polynomial(
+      make_not_null(&spec_functions_of_time_third_order), function_of_time_file,
+      function_of_time_name_map);
+
+  bool uses_quaternion_rotation = false;
+  for (const auto& name_and_fot : *functions_of_time) {
+    auto* maybe_quaternion_fot =
+        dynamic_cast<domain::FunctionsOfTime::QuaternionFunctionOfTime<3>*>(
+            name_and_fot.second.get());
+    if (maybe_quaternion_fot != nullptr) {
+      uses_quaternion_rotation = true;
+    }
+  }
+
+  // Only parse as quaternion function of time if it exists
+  if (uses_quaternion_rotation) {
+    domain::FunctionsOfTime::read_spec_piecewise_polynomial(
+        make_not_null(&spec_functions_of_time_quaternion),
+        function_of_time_file, function_of_time_name_map, true);
+  }
+
+  for (const auto& [spec_name, spectre_name] : function_of_time_name_map) {
+    (void)spec_name;
+    // The FunctionsOfTime we are mutating must already have
+    // an element with key==spectre_name; this action only
+    // mutates the value associated with that key
+    if (functions_of_time->count(spectre_name) == 0) {
+      ERROR("Trying to import data for key "
```

Comment:
Can this and the error below have error tests added fairly easily? It would be nice to gradually increase coverage of our errors